Gems/Atom/RHI/Vulkan/Code/Source/RHI/SwapChain.cpp

```diff
@@ -483,20 +483,26 @@ namespace AZ
         void SwapChain::InvalidateNativeSwapChain()
         {
             auto& device = static_cast<Device&>(GetDevice());
-            vkDeviceWaitIdle(device.GetNativeDevice());
-            if (m_nativeSwapChain != VK_NULL_HANDLE)
+            auto presentCommand = [this,  &device]([[maybe_unused]] void* queue)
             {
-                vkDestroySwapchainKHR(device.GetNativeDevice(), m_nativeSwapChain, nullptr);
-                m_nativeSwapChain = VK_NULL_HANDLE;
-            }
+                vkDeviceWaitIdle(device.GetNativeDevice());
+                if (m_nativeSwapChain != VK_NULL_HANDLE)
+                {
+                    vkDestroySwapchainKHR(device.GetNativeDevice(), m_nativeSwapChain, nullptr);
+                    m_nativeSwapChain = VK_NULL_HANDLE;
+                }
+            };
+
+            m_presentationQueue->QueueCommand(AZStd::move(presentCommand));
+            m_presentationQueue->FlushCommands();
         }
 
         RHI::ResultCode SwapChain::CreateSwapchain()
         {
             auto& device = static_cast<Device&>(GetDevice());
 
             m_surfaceCapabilities = GetSurfaceCapabilities();
-            m_surfaceFormat = GetSupportedSurfaceFormat(GetDescriptor().m_dimensions.m_imageFormat);
+            m_surfaceFormat = GetSupportedSurfaceFormat(m_dimensions.m_imageFormat);
```

Comment:
I noticed this issue too. I am thinking to remove m_dimensions completely from swapchain class, since it has no difference to the one stored in descriptor.
However in the current case, it will test success before the descriptor is copied. We can reorder the steps by moving the set descriptor earlier, and if failure, clear it.
https://github.com/o3de/o3de/blob/56904d17991d866d818ec0c773e392c541634b04/Gems/Atom/RHI/Code/Source/RHI/SwapChain.cpp#L57

Reply:
Yeah that works.